src/os/bluestore/BlueStore.cc

```diff
@@ -13220,7 +13220,7 @@ int BlueStore::_do_alloc_write(
       // do an approximate (fast) estimation for resulting blob size
       // that doesn't take header overhead  into account
       uint64_t result_len = p2roundup(compressed_len, min_alloc_size);
-      if (r == 0 && result_len <= want_len && result_len < wi.blob_length) {
+      if (r == 0 && result_len <= want_len && result_len <= wi.blob_length) {
```

Comment:
i am not sure if we should accept this. as it is likely a false positive that the compression rate is acceptable, even though the compression rate is 100%.

i understand that we might reject the test even if it fulfill the required ratio, if the bluestore_compression_required_ratio is close to 100%. but i don't think this is the correct fix. probably we should use a greater "compression_max_blob_size" in this case?

Reply:
yeah,greater compression_max_blob_size may lead to more compression.